clustered/client/src/main/java/org/ehcache/clustered/client/internal/loaderwriter/writebehind/ClusteredWriteBehindStore.java

```diff
@@ -0,0 +1,385 @@
+import org.ehcache.clustered.client.internal.store.operations.codecs.OperationsCodec;
+import org.ehcache.clustered.client.service.ClusteringService;
+import org.ehcache.clustered.common.internal.store.Chain;
+import org.ehcache.clustered.common.internal.store.Element;
+import org.ehcache.config.ResourceType;
+import org.ehcache.core.spi.store.tiering.AuthoritativeTier;
+import org.ehcache.core.spi.time.TimeSource;
+import org.ehcache.core.spi.time.TimeSourceService;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
+import org.ehcache.spi.loaderwriter.WriteBehindConfiguration;
+import org.ehcache.spi.resilience.StoreAccessException;
+import org.ehcache.spi.service.ServiceConfiguration;
+import org.ehcache.spi.service.ServiceDependencies;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeoutException;
+
+import static org.ehcache.core.exceptions.StorePassThroughException.handleException;
+import static org.ehcache.core.spi.service.ServiceUtils.findSingletonAmongst;
+
+public class ClusteredWriteBehindStore<K, V> extends ClusteredStore<K, V> implements AuthoritativeTier<K, V> {
+
+  private final CacheLoaderWriter<? super K, V> cacheLoaderWriter;
+  private final ClusteredWriteBehind clusteredWriteBehind;
+
+  private ClusteredWriteBehindStore(Configuration<K, V> config,
+                                    OperationsCodec<K, V> codec,
+                                    ChainResolver<K, V> resolver,
+                                    TimeSource timeSource,
+                                    CacheLoaderWriter<? super K, V> loaderWriter,
+                                    ExecutorService executorService) {
+    super(config, codec, resolver, timeSource);
+    this.cacheLoaderWriter = loaderWriter;
+    this.clusteredWriteBehind = new ClusteredWriteBehind(executorService,
+                                                         timeSource,
+                                                         resolver,
+                                                         this.cacheLoaderWriter,
+                                                         codec);
+  }
+
+  private LockManager getProxy() {
+    return (LockManager) storeProxy;
+  }
+
+  @Override
+  protected ValueHolder<V> getInternal(K key) throws StoreAccessException, TimeoutException {
+    try {
+      Chain chain = storeProxy.get(extractLongKey(key));
+      if (!chain.isEmpty()) {
+        ClusteredValueHolder<V> holder = null;
+        ResolvedChain<K, V> resolvedChain = resolver.resolve(chain, key, timeSource.getTimeMillis());
+        Result<K, V> resolvedResult = resolvedChain.getResolvedResult(key);
+        if (resolvedResult != null) {
+          V value = resolvedResult.getValue();
+          long expirationTime = resolvedChain.getExpirationTime();
```

Comment:
why do we need to handle expiry logic here, AFAIK the resolver should be able to handle all that

Reply:
This code is similar to getInternal() in ClusteredStore.